packages/grid/x-data-grid/src/components/GridRow.tsx

```diff
@@ -277,12 +277,15 @@ const GridRow = React.forwardRef<HTMLDivElement, GridRowProps>(function GridRow(
       'width' | 'colSpan' | 'showRightBorder' | 'indexRelativeToAllColumns'
     >,
   ) => {
-    const disableDragEvents =
-      (disableColumnReorder && column.disableReorder) ||
-      (!rowReordering &&
-        !!sortModel.length &&
-        treeDepth > 1 &&
-        Object.keys(editRowsState).length > 0);
+    // when the cell is a reorder cell we are not allowing to reorder the col
+    // fixes https://github.com/mui/mui-x/issues/11126
+    const isReorderCell = column.field === '__reorder__';
+    const isEditingRows = Object.keys(editRowsState).length > 0;
+
+    const canReorderColumn = !(disableColumnReorder || column.disableReorder);
+    const canReorderRow = rowReordering || !sortModel.length || treeDepth <= 1 || isEditingRows;
```

Comment:
Is this expression correct? I would expect it to be:

```tsx
 const canReorderRow = rowReordering && !sortModel.length && treeDepth <= 1 && !isEditingRows;
 ```

Reply:
nice catch ... it was working like this as well ... but the cases where it shouldn't be working did work as well, so that would have been bad! Thanks for catching that!